app/models/course.rb

```diff
@@ -53,6 +53,13 @@ class Course < ActiveRecord::Base
   accepts_nested_attributes_for :invitations, :assessment_categories
 
   scope :ordered_by_title, -> { order(:title) }
+  scope :ordered_by_end_at, ->(direction = :desc) { order(end_at: direction) }
+
+  # @!method with_user
+  #   Selects all the courses with user as one of its members
+  scope :with_user, (lambda do |user|
+    joins { course_users }.where { course_users.user_id == my { user.id } }
```

Comment:
We don't need the `my` here, I think that is used if we want to access the object's instance methods. 

Since `user.id` is an attribute, I think `joins { course_users }.where { course_users.user_id == user.id }` will suffice.